packages/twenty-server/src/engine/core-modules/open-api/utils/components.utils.ts

```diff
@@ -60,14 +61,14 @@ const getSchemaComponentsProperties = (
       case FieldMetadataType.ADDRESS:
         itemProperty = {
           type: 'object',
-          properties: Object.keys(field.targetColumnMap).reduce(
-            (properties, key) => {
-              properties[key] = { type: 'string' };
+          properties: compositeTypeDefintions
+            .get(field.type)
+            ?.properties?.reduce((properties, property) => {
+              // TODO: This should not be statically typed, instead we should do someting recursive
+              properties[property.name] = { type: 'string' };
```

Comment:
@martmull Is it normal to always type composite type as string, because they can have other kind of type

Reply:
It is not normal. But as the types are not stored anywhere with the previous TargetColumnMap implementation, we defaulted it as `string`